app/v1/product/data/ProductRepository.scala

```diff
@@ -0,0 +1,113 @@
+import play.api.{Logger, MarkerContext}
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+case class ProductData(id: Int, id_typeProduct: Int, name: String, gender: String, size: String, price: Double)
+
+case class TypeProduct(id_typeProduct: Int, name: String)
+
+case class ManageTypeProduct @Inject()(connection: Connection) {
+
+  def create(data: TypeProduct): doobie.Update0 = {
+    sql"insert into typeProduct (id_typeProduct, name) values (${data.id_typeProduct},${data.name})".update
+  }
+
+  def list(): Iterable[TypeProduct] = {
+    val typeProductListDB = sql"select * from typeproduct".query[TypeProduct].to[List]
+    connection.transactor.use(typeProductListDB.transact[IO]).unsafeRunSync()
+  }
+
+  def get(id: Int): TypeProduct = {
+    val typeProductDB = sql"""select * from typeproduct where id_typeproduct = $id""".query[TypeProduct].unique
+    connection.transactor.use(typeProductDB.transact[IO]).unsafeRunSync()
+  }
+}
+
+class ProductExecutionContext @Inject()(actorSystem: ActorSystem)
+  extends CustomExecutionContext(actorSystem, "repository.dispatcher")
+
+/**
+  * A pure non-blocking interface for the ProductRepository.
+  */
+trait ProductRepository {
+  def list()(implicit mc: MarkerContext): Future[Iterable[ProductData]]
+
+  def get(id: Int)(implicit mc: MarkerContext): Future[Option[ProductData]]
+
+  def create(data: ProductData)(implicit mc: MarkerContext): Future[Int]
+
+  def delete(id: Int)(implicit mc: MarkerContext): Future[Iterable[ProductData]]
+}
+
+class ProductRepositoryImpl @Inject()()(connection: Connection)(implicit ec: ProductExecutionContext)
+  extends ProductRepository {
+
+  private val logger = Logger(this.getClass)
+
+  val startDataTable: Unit = {
+
+    val productList = List(
+      ProductData(1, 1, "Shirt", "W", "M", 30.5),
+      ProductData(2, 1, "Skirt", "W", "L", 32),
+      ProductData(3, 2, "Boots", "M", "42", 45.99),
+      ProductData(4, 1, "Shirt", "M", "XL", 35.99),
+      ProductData(5, 3, "Hat", "W", "s", 59.99),
+      ProductData(6, 2, "Sneakers", "W", "41", 47.5),
+    )
+
+    val truncatedTable = sql"truncate table product".update.run
+
+    val inserts = productList.flatMap(p => s"(${p.id}, ${p.id_typeProduct}, \'${p.name}\', \'${p.gender}\', " +
+      s"\'${p.size}\', ${p.price})").mkString.replace(")(", "),\n\t(")
+    val startTable = (fr"insert into product (id_product, id_typeProduct, name, gender, size, price) values"
+      ++ Fragment.const(inserts)).update.run
+
+    val l = connection.transactor.use((truncatedTable, startTable).mapN(_ + _).transact[IO]).unsafeRunSync()
+    logger.trace(s"inserting initial rows(${l}):\n\t$inserts")
+  }
+
+  override def list()(implicit mc: MarkerContext): Future[Iterable[ProductData]] = {
+    Future {
+      logger.trace("list: ")
+      val productList = sql"select * from product".query[ProductData].to[List]
+      connection.transactor.use(productList.transact[IO]).unsafeRunSync()
```

Comment:
Get rid of that `Future` blocks.

```suggestion
      connection.transactor.use(productList.transact[IO]).unsafeToFuture()
```

Reply:
for every method from trait ProductRepository??

Reply:
done in commit fe8a098

Reply:
@vejeta Now, I understand what you asked about Roberto's request.
It is unnecessary to put the result inside an artificially created Future{}

So, just need to leave the code outside of the Future block, removing the Future.


A question for @bilki, is it a common pattern to use unsafeRunSync() on this layer?